bin/api_diff.ml

```diff
@@ -1,21 +1,13 @@
-let _ = Api_watch_diff.diff_interface ~reference:"reference" ~current:"current"
-
 let run (`Ref_cmi reference) (`Current_cmi current) =
   let current = Cmi_format.read_cmi current in
   let reference = Cmi_format.read_cmi reference in
-  let typing_env = Env.empty in
-  let coercion1 () =
-    Includemod.signatures typing_env ~mark:Mark_both reference.cmi_sign
-      current.cmi_sign
-  in
-  let coercion2 () =
-    Includemod.signatures typing_env ~mark:Mark_both current.cmi_sign
-      reference.cmi_sign
+  let coercion =
+    Api_watch_diff.diff_interface ~reference:reference.cmi_sign
+      ~current:current.cmi_sign
   in
-  match (coercion1 (), coercion2 ()) with
-  | Tcoerce_none, Tcoerce_none -> Printf.printf "API unchanged!\n"
-  | _, _ -> Printf.printf "API changed!\n"
-  | exception Includemod.Error _ -> Printf.printf "API changed!\n"
+  match coercion with
+  | false -> Printf.printf "API unchanged!\n"
+  | true -> Printf.printf "API changed!\n"
```

Comment:
Since you are matching over a boolean here, note that you could rewrite this using an if-then-else:
```suggestion
  if differs then
    Printf.printf "API changed!\n"
  else
    Printf.printf "API unchanged!\n"
```

That's up to you, I don't personally mind the pattern-matching here!

Reply:
> Since you are matching over a boolean here, note that you could rewrite this using an if-then-else:
@NathanReb Yeah,I was thinking about doing that but since the project was already using pattern matching for printing, I didn't want to make unnecessary changes. I will do it now.